test/byContract/IexecEscrow/IexecEscrowNative.ts

```diff
@@ -0,0 +1,280 @@
+                    iexecPocoAsAccountA.depositForArray(...depositForArrayArgs),
+                ).to.be.revertedWith('invalid-array-length');
+            });
+        });
+
+        describe('Withdrawals', () => {
+            it('Should withdraw native tokens', async () => {
+                await iexecPocoAsAccountA.deposit(...depositArgs);
+
+                expect(await iexecPocoAsAccountA.callStatic.withdraw(...withdrawArg)).to.be.true;
+                await expect(iexecPocoAsAccountA.withdraw(...withdrawArg))
+                    .to.changeEtherBalances(
+                        [accountA, iexecPoco],
+                        [depositNativeAmount, -depositNativeAmount],
+                    )
+                    .to.changeTokenBalances(iexecPoco, [accountA], [-withdrawAmount])
+                    .to.emit(iexecPoco, 'Transfer')
+                    .withArgs(accountA.address, AddressZero, withdrawAmount);
+            });
+
+            it('Should withdraw native tokens to another address', async () => {
+                await iexecPocoAsAccountA.deposit(...depositArgs);
+                const withdrawToArgs = [...withdrawArg, accountB.address] as [BigNumber, string];
+                expect(await iexecPocoAsAccountA.callStatic.withdrawTo(...withdrawToArgs)).to.be
+                    .true;
+                await expect(iexecPocoAsAccountA.withdrawTo(...withdrawToArgs))
+                    .to.changeEtherBalances(
+                        [accountB, iexecPoco],
+                        [depositNativeAmount, -depositNativeAmount],
+                    )
+                    .to.changeTokenBalances(iexecPoco, [accountA], [-withdrawAmount])
+                    .to.emit(iexecPoco, 'Transfer')
+                    .withArgs(accountA.address, AddressZero, withdrawAmount);
+            });
+
+            it('Should not withdraw native tokens with empty balance', async () => {
+                await expect(
+                    iexecPocoAsAccountA.withdraw(...withdrawArg),
+                ).to.be.revertedWithoutReason();
+            });
+
+            it('Should not withdraw native tokens with insufficient balance', async () => {
+                await iexecPocoAsAccountA.deposit(...depositArgs);
+
+                await expect(
+                    iexecPocoAsAccountA.withdraw(depositAmount.mul(2)),
+                ).to.be.revertedWithoutReason();
+            });
+        });
+
+        describe('Recover', () => {
+            it('Should recover extra balance', async () => {
+                await iexecAdmin.sendTransaction({
+                    to: iexecPoco.address,
+                    value: depositNativeAmount,
+                });
+                await setStorageAt(
+                    proxyAddress,
+                    '0x0c', // Slot index of `m_totalSupply` in Store
+                    depositAmount.div(2).toHexString(),
+                );
+                const expectDelta = depositAmount.div(2).toNumber();
```

Comment:
```suggestion
                const totalSupply = await iexecPoco.totalSupply();
                const expectedDelta = BigNumber.from(5);
                await setStorageAt(
                    proxyAddress,
                    '0x0c', // Slot index of `m_totalSupply` in Store
                    totalSupply.sub(expectDelta).toHexString(),
                );
```

Reply:
And also assert `iexecPoco.totalSupply()` before and after the storage override to make sure its value changes.